packages/core/src/v3/runMetadata/metadataStream.ts

```
@@ -0,0 +1,88 @@
+  key: string;
+  iterator: AsyncIterator<T>;
+  signal?: AbortSignal;
+};
+
+export class MetadataStream<T> {
+  private controller = new AbortController();
+  private serverQueue: Array<Promise<IteratorResult<T>>> = [];
+  private consumerQueue: Array<Promise<IteratorResult<T>>> = [];
+  private serverIterator: AsyncIterator<T>;
+  private consumerIterator: AsyncIterator<T>;
+  private streamPromise: Promise<void | Response>;
```

Biome

[error] void is confusing inside a union type.

Unsafe fix: Use undefined instead.


(lint/suspicious/noConfusingVoidType)